solr/core/src/java/org/apache/solr/handler/admin/api/ListAliasesAPI.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.solr.response.SolrQueryResponse;
+
+/** V2 APIs for managing and inspecting collection aliases */
+@Path("/aliases")
+public class ListAliasesAPI extends AdminAPIBase {
+
+  @Inject
+  public ListAliasesAPI(
+      CoreContainer coreContainer,
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse) {
+    super(coreContainer, solrQueryRequest, solrQueryResponse);
+  }
+
+  /**
+   * V2 API for listing all aliases known by Solr.
+   *
+   * <p>This API <code>GET /api/aliases</code> is analogous to the v1 <code>GET /api/cluster/aliases
+   * </code> API.
+   */
+  @GET
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @PermissionName(COLL_READ_PERM)
+  @Operation(
+      summary = "List the existing collection aliases.",
+      tags = {"aliases"})
+  public ListAliasesResponse getAliases() throws Exception {
+    recordCollectionForLogAndTracing(null, solrQueryRequest);
+
+    final ListAliasesResponse response = instantiateJerseyResponse(ListAliasesResponse.class);
+    final CoreContainer coreContainer = fetchAndValidateZooKeeperAwareCoreContainer();
+
+    ZkStateReader zkStateReader = coreContainer.getZkController().getZkStateReader();
+
+    // if someone calls listAliases, lets ensure we return an up to date response
+    zkStateReader.getAliasesManager().update();
+
+    Aliases aliases = zkStateReader.getAliases();
+
+    if (aliases != null) {
+      // the aliases themselves...
+      response.aliases = aliases.getCollectionAliasMap();
+      // Any properties for the above aliases.
+      Map<String, Map<String, String>> meta = new LinkedHashMap<>();
+      for (String alias : aliases.getCollectionAliasListMap().keySet()) {
+        Map<String, String> collectionAliasProperties = aliases.getCollectionAliasProperties(alias);
+        if (!collectionAliasProperties.isEmpty()) {
+          meta.put(alias, collectionAliasProperties);
+        }
+      }
+      response.properties = meta;
+    }
+
+    return response;
+  }
+
+  @GET
+  @Path("/{aliasName}")
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @PermissionName(COLL_READ_PERM)
+  @Operation(
+      summary = "Get details for a specific collection alias.",
+      tags = {"aliases"})
+  public GetAliasByNameResponse getAliasByName(
+      @Parameter(description = "Alias name.", required = true) @PathParam("aliasName")
+          String aliasName)
+      throws Exception {
+    recordCollectionForLogAndTracing(null, solrQueryRequest);
+
+    final GetAliasByNameResponse response = instantiateJerseyResponse(GetAliasByNameResponse.class);
+    response.alias = aliasName;
+
+    final CoreContainer coreContainer = fetchAndValidateZooKeeperAwareCoreContainer();
+    ZkStateReader zkStateReader = coreContainer.getZkController().getZkStateReader();
+    // if someone calls listAliases, lets ensure we return an up to date response
```

Comment:
[-0] Small nit: copy/paste issue with 'listAliases'.

Relatedly though, maybe this code should be pulled into a private helper method that both `getAliasByName` and `getAliases` can call, to cut down on duplication.

Reply:
looks good